crates/dojo-erc/src/erc1155/components.cairo

```diff
@@ -0,0 +1,28 @@
+use starknet::ContractAddress;
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct OperatorApproval {
+    #[key]
+    token: ContractAddress,
+    #[key]
+    owner: ContractAddress,
+    #[key]
+    operator: ContractAddress,
+    approved: bool
+}
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Uri {
+    #[key]
+    token: ContractAddress,
+    uri: felt252
+}
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Balance {
+    #[key]
+    token: ContractAddress,
```

Comment:
I think this is missing a `token_id #[key]`, as things stand it's the same as the ERC20 balance.